variables.tf

```diff
@@ -107,3 +107,9 @@ variable "auth_enabled" {
   type        = bool
   default     = false
 }
+
+variable "transit_encryption_mode" {
+  description = "The TLS mode of the Redis instance, If not provided, TLS is disabled for the instance."
+  type        = string
+  default     = "DISABLED"
```

Comment:
This seems like something we should enable by default with a breaking release although there are some [performance caveats](https://cloud.google.com/memorystore/docs/redis/in-transit-encryption#performance_impact_of_enabling_in-transit_encryption). 
/cc @morgante

Reply:
I agree, we should cut a breaking change for this. It looks more secure.

Reply:
I think that's a tough call. Since I don't believe there is a way currently to get to SSL keys via terraform or anything, so it limits a 100% setup. At least the auth_string comes back and can be then applied to other resources.

Reply:
Enable redis_auth as well?

Reply:
Probably not. For auth, I think IAM is better.

Reply:
This feature requires the Certificate Authorities (CA) that are used to verify the identity of the server, see: https://cloud.google.com/memorystore/docs/redis/in-transit-encryption 

I think the output should [include the cert](https://cloud.google.com/memorystore/docs/redis/enabling-in-transit-encryption#downloading_the_certificate_authority) so I can say, save it in Secret Manager for use by a VM or Container.

I might be able to fork and implement the CA feature at some future time, but not for a few weeks at least. Should I open a new issue or PR to change the default to "DISABLED" for now?

Thank you for your work on the modules by the way, they have been very helpful!

Reply:
Open an issue please.. i might be able to get to it today

Reply:
https://github.com/terraform-google-modules/terraform-google-memorystore/pull/99

Reply:
Wow, blown away by the help. Let me know if I can contribute anything to follow up (docs, testing). Cheers!